inbox/models/folder.py

```diff
@@ -36,9 +36,9 @@ class Folder(MailSyncBase, UpdatedAtMixin, DeletedAtMixin):
     # NOTE: this doesn't hold for EAS, which is case insensitive for non-Inbox
     # folders as per
     # https://msdn.microsoft.com/en-us/library/ee624913(v=exchg.80).aspx
-    name = Column(String(MAX_FOLDER_NAME_LENGTH, collation='utf8mb4_bin'),
-                  nullable=False)
-    canonical_name = Column(String(MAX_FOLDER_NAME_LENGTH), nullable=False,
+    name = Column(CategoryNameType(MAX_INDEXABLE_LENGTH,
+                  collation='utf8mb4_bin'), nullable=False)
+    canonical_name = Column(String(MAX_INDEXABLE_LENGTH), nullable=False,
```

Comment:
It seems that `canonical_name` for Folder/Label and `name` for Category are never truncated, probably due to the fact that they corresponds to the `role`, which can never be that long. I wasn't sure if adding truncation to these fields was necessary or would cause any trouble, and given that we never have truncated it I left it as so.